src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CachedCaseUserRepository.java

```diff
@@ -19,32 +19,39 @@
 
     public static final String QUALIFIER = "cached";
 
-    private final Map<String, List<Long>> casesUserHasAccess = newHashMap();
-    private final Map<String, List<String>> caseUserRoles = newHashMap();
-
     public CachedCaseUserRepository(@Qualifier(DefaultCaseUserRepository.QUALIFIER)
-                                        CaseUserRepository caseUserRepository) {
+                                    final CaseUserRepository caseUserRepository) {
         this.caseUserRepository = caseUserRepository;
     }
 
     @Override
+    @Caching(evict = {
+        @CacheEvict(value = "casesForUserCache", key = "#userId", allEntries = true),
```

Comment:
Does it work correctly with cluster of nodes? As other nodes may update records in parallel.

Reply:
It looks like the embedded cache topology will not evict the caches from cluster of instances. We might need client-server topology which has its own drawback of network communication etc. So, we have decided to stick with embedded cache topology by considering the nature of our application.